File: core/src/main/java/com/views/imageview/DynamicHeightKenBurnsView.java
package com.views.imageview;

import android.content.Context;
import android.util.AttributeSet;

import com.flaviofaria.kenburnsview.KenBurnsView;

//TODO convert kotlin
public class DynamicHeightKenBurnsView extends KenBurnsView {
    private float mAspectRatio = 1.5f;

    public DynamicHeightKenBurnsView(Context context) {
        super(context);
    }

    public DynamicHeightKenBurnsView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public DynamicHeightKenBurnsView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public void setAspectRatio(float aspectRatio) {
        mAspectRatio = aspectRatio;
        requestLayout();
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        int measuredWidth = getMeasuredWidth();
        setMeasuredDimension(measuredWidth, (int) (measuredWidth / mAspectRatio));
    }
}

File: core/src/main/java/com/loitpcore/views/imageView/DynamicHeightKenBurnsView.kt
package com.loitpcore.views.imageView

import android.content.Context
import android.util.AttributeSet
import com.flaviofaria.kenburnsview.KenBurnsView

class DynamicHeightKenBurnsView : KenBurnsView {
    private var mAspectRatio = 1.5f

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(
        context,
        attrs,
        defStyle
    )

    fun setAspectRatio(aspectRatio: Float) {

        mAspectRatio = aspectRatio
        requestLayout()
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)

        val measuredWidth = measuredWidth
        setMeasuredDimension(measuredWidth, (measuredWidth / mAspectRatio).toInt())
    }
}
